### DISCUSSION TITLE
adding my specific auth token

### DISCUSSION THREAD

**Discussion by DannyDarth** [streamlink/streamlink#5005] [Q&A] [answered]
I was reading the documents on using my specific token to a stream. Is there a way to add it to the streamlink twitch gui settings? Hopefully i am asking in the right place.

**Comment by mkbloke** [comment]
I don't know anything about the Twitch GUI, but it sounds like you probably want to use a [plugin specific configuration file](https://streamlink.github.io/cli/config.html#plugin-specific-configuration-file).

**Comment by bastimeyer** [comment, marked as answer]
1. Wrong place, this is the Streamlink repo, not Streamlink Twitch GUI
2. Already answered here: https://github.com/streamlink/streamlink-twitch-gui/issues/754

**Reply by NeoFahrenheit** [reply]
Hi, I have read those docs and some more, but I can't find the information where to put my Twitch auth-token in the streamlink python package. The **--http-header** or --**twitch-api-header** seems to be CLI only. I have tried this, but Streamlink class does not seems to accept it.

```
class MyClass:
    def __init__(self):
        auth = 'abcdefghijklmnopqrstuvwxyz0123'
        self.session = streamlink.Streamlink(options={'--twitch-api-header': f"Authorization=OAuth {auth}"})
```

Can you help?
Thank you.

**Reply by bastimeyer** [reply]
Please don't hijack unrelated discussion posts with different questions.

> `options={'--twitch-api-header': ...}`

session options != CLI parameter names
session options != plugin options

https://streamlink.github.io/api.html#streamlink.Streamlink

```py
session = Streamlink()
session.set_plugin_option("twitch", "api-header", [("Authorization", "OAuth TOKEN_HERE")])
```

**Comment by DannyDarth** [comment]
I guess i dont know where to make or add the config file for twitch gui.
